### ISSUE TITLE
Migration: Pass Chart elements by ref / ElementReference (not by id) to JSInterop

### ISSUE THREAD

**Issue by SeppPenner** [mariusmuntean/ChartJs.Blazor#42]
Migrated from https://github.com/Joelius300/ChartJSBlazor/issues/76:

### Describe the feature request
It would be a good idea to use  the `ElementReference` type to reference the charts from C# to JS. Example (Although it's a bit outdated): https://visualstudiomagazine.com/blogs/tool-tracker/2018/11/an-ad-hoc-approach.aspx

### Which charts does this feature request apply to?
All.

### Describe the solution you'd like
Like in the example:

Razor:

```razor
<input type="text" ref="mytextbox" value="Peter" />

@code
{
    ElementReference mytextbox;

    protected override Task OnInitializedAsync()
    {
        await JSRuntime.Current.InvokeAsync<string>("SetName", mytextbox);
    }
}
```

Javascript:

```js
function SetName(textbox) {
    textbox.value = "Vogel";
}
```
### Describe alternatives you've considered
None.

### Additional context
Example: https://visualstudiomagazine.com/blogs/tool-tracker/2018/11/an-ad-hoc-approach.aspx.

**Comment by Joelius300** [comment]
Thank you but I'm going to close this as wontfix. I generally like the idea but I have 1.5 reasons to decline this:

- We're working with a weird setup of double json serialization at the moment because [we still can't pass serializer options to IJSRuntime](https://github.com/dotnet/aspnetcore/issues/12685). These reference objects exist both in C# and in JavaScript and in order for them to behave correctly, they need to be passed through the serializer along with a special converter which is registed when using any `JSRuntime`. Since we mangle the object to an `ExpandoObject` before it even reaches that serializer, the reference won't work.
- The current way works great and from a performance standpoint I don't think you'll ever see a difference.

In other words: it would require extra attention to make it work and the current solution has no known downsides.
